### PR TITLE
Add constants for reserve modes

### LINKED ISSUE #172: stdlib: Introduce constants for modes of `nativeReserve()`
Analogously to the [constants](https://github.com/tact-lang/tact/blob/main/stdlib/std/send.tact#L4) for the [message mode](https://docs.tact-lang.org/book/message-mode) of `SendParameters` struct, a short series of constants may be introduced for the `mode` parameter of the [`nativeReserve()`](https://github.com/tact-lang/tact/blob/main/stdlib/std/reserve.tact#L2).

Description of possible values is already [present](https://docs.tact-lang.org/language/ref/advanced#nativereserve) in the Tact documentation (though poorly formatted as of now, I'll fix it soon).

### PR DESCRIPTION
**Pull request by Gusarich**
closes #172

### CODE REVIEW

**Comment by Gusarich** [comment]
@novusnota also would be nice to see a better explanation on these modes in docs, similarly to what we have for SendParamters

**Comment by novusnota** [comment]
@Gusarich sure, right after version 1.3.0 gets released

**Review by anton-trunov — APPROVED** [review]

**Review by anton-trunov — CHANGES_REQUESTED** [review]
Wait, there are three base modes and three flags which are supposed to modify the base mode behavior. I'd like to have this distinction in the constant names too.

File: stdlib/std/reserve.tact [hunk 1]
```diff
@@ -1,2 +1,9 @@
...
-native nativeReserve(amount: Int, mode: Int);
\ No newline at end of file
+native nativeReserve(amount: Int, mode: Int);
+
+const ReserveExact: Int = 0;
+const ReserveAllExcept: Int = 1;
+const ReserveAtMost: Int = 2;
+const ReserveAddOriginalBalance: Int = 4;
+const ReserveInvertSign: Int = 8;
+const ReserveBounceIfActionFail: Int = 16;
```
Comment (anton-trunov) [inline stdlib/std/reserve.tact:9]: Currently, neither TVM docs, not Tact docs define the +16 flag (`ReserveBounceIfActionFail`):

image

Where does it come from?

Comment (Gusarich) [reply stdlib/std/reserve.tact:9]: Screenshot 2024-03-25 at 8 08 14 PM
https://docs.ton.org/learn/tvm-instructions/tvm-upgrade-2023-07#sending-messages

Comment (anton-trunov) [reply stdlib/std/reserve.tact:9]: Got it, thanks!

**Review by anton-trunov — APPROVED** [review]
After an internal discussion, looks like we can merge this without making a distinction between base modes and flags changing their behavior

**Comment by tonMaxi** [comment]
wowo

**Pull request merged**